equistore/io.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+
+from ._c_lib import _get_library
+from .block import TensorBlock
+from .data import _is_numpy_array, _is_torch_array
+from .labels import Labels
+from .tensor import TensorMap
+
+
+def load(path: str, use_numpy=False) -> TensorMap:
+    """Load a previously saved :py:class:`equistore.TensorMap` from the given path.
+
+    :py:class:`equistore.TensorMap` are serialized using numpy's ``.npz``
+    format, i.e. a ZIP file without compression (storage method is ``STORED``),
+    where each file is stored as a ``.npy`` array. See the C API documentation
+    for more information on the format.
+
+    :param path: path of the file to load
+    :param use_numpy: should we use numpy or the native implementation? The main
+        difference is that numpy should be able to save more dtypes for the data
+        than the native implementation, which is limited to float64.
+    """
+    if use_numpy:
+        return _read_npz(path)
+    else:
+        lib = _get_library()
+        return TensorMap._from_ptr(lib.eqs_tensormap_load(path.encode("utf8")))
+
+
+def save(path: str, tensor: TensorMap, use_numpy=False):
+    """Save the given :py:class:`equistore.TensorMap` to a file at ``path``.
+
+    :py:class:`equistore.TensorMap` are serialized using numpy's ``.npz``
+    format, i.e. a ZIP file without compression (storage method is ``STORED``),
+    where each file is stored as a ``.npy`` array. See the C API documentation
+    for more information on the format.
+
+    :param path: path of the file where to save the data
+    :param tensor: tensor to save
+    :param use_numpy: should we use numpy or the native implementation? The main
```

Comment:
why is `use_numpy` false by default. The docstring sounds like it is beneficial to use numpy.

Reply:
The non-numpy version should be slightly faster since it construct Rust struct directly instead of building blocks in Python first (although I did not benchmark this). Also, using the non-numpy version by default allows to find bugs/issues in the native version faster =)

Reply:
Okay then we should maybe add the higher perfomance of the native implementation to the docstring. Python users are used to the fact that numpy is the fastest you can do :-).

Reply:
Really cool. So we should add this information to the docs and motivate people use the native format.

Reply:
I've added a small sentence on this, it should be good to go =)